packages/js/api-core-tests/jest.config.js

```diff
@@ -20,4 +20,22 @@ module.exports = {
 
 	// Indicates whether each individual test should be reported during the run
 	verbose: verboseOutput,
+
+	/**
+	 * Configure `jest-allure` for Jest v24 and above.
+	 *
+	 * @see https://www.npmjs.com/package/jest-allure#jest--v-24-
+	 */
+	setupFilesAfterEnv: [
+		'jest-allure/dist/setup',
+		'<rootDir>/allure.config.js',
+	],
+
+	/**
+	 * Make sure Jest is using Jasmine as its test runner.
+	 * `jest-allure` does not work with the `jest-circus` test runner, which is the default test runner of Jest starting from v27.
+	 *
+	 * @see https://github.com/zaqqaz/jest-allure#uses-jest-circus-or-jest--v-27-
+	 */
+	testRunner: 'jasmine2',
```

Comment:
Ouch... It's a real bummer that the Allure integration does not work with `jest-circus`. I understand this change is going to be limited only to the API package tests, not e2e-environment/woocommerce tests, correct?

Good read: https://github.com/facebook/jest/pull/3668

Reply:
Hi @Luc45 thanks for asking this question!

> this change is going to be limited only to the API package tests, not e2e-environment/woocommerce tests, correct?

Yes, that is correct. Only the Core API tests will use `jasmine` as the testRunner, not the e2e ones. @MrJnrman correctly specified the testRunner to `jest-circus` [here](https://github.com/woocommerce/woocommerce/blob/b30179f8d234e629860c4f253a78ee5f2816186a/packages/js/e2e-environment/config/jest.config.js#L59) for the e2e tests.

> It's a real bummer that the Allure integration does not work with jest-circus

> Good read: https://github.com/facebook/jest/pull/3668

True! And thanks for suggesting to read that FB PR! 

Also, Jest is planning to remove `jasmine` on their _"next major version"_ as they mentioned in their [Jest 27: New Defaults for Jest, 2021 edition](https://jestjs.io/blog/2021/05/25/jest-27). To quote:

> In the next major, we plan to also eliminate jest-jasmine2 and jest-environment-jsdom from the Jest dependency tree and require them to be installed explicitly...

So for now, we got three options:
1. Hope that `jest-allure` would support `jest-circus` as the testRunner in the future.
2. Stick to Jest v27 or older (currently we're using v25)
3. If we found it necessary to upgrade our Core API tests to Jest v28 in the future, we will have to import the jest-jasmine2 package explicitly.

But I think we can address these once we get to that point.